Historically, <sys/mman.h> did not define madvise() or getpagesizes()
properly for C++, leading mysql to compensate by defining it directly
in the code. Since then, <sys/mman.h> has been fixed. Removing this now
allows us to get across the transition from the obsolete caddr_t
definition to the modern (void *). See:

	29897679 ON headers use caddr_t for standard interfaces
		that should be (void *)

--- mysql-5.7.25/sql/mysqld.cc.orig	2018-12-21 10:39:40.000000000 +0000
+++ mysql-5.7.25/sql/mysqld.cc	2019-06-15 06:00:15.497204482 +0000
@@ -159,11 +159,6 @@
 
 #define mysqld_charset &my_charset_latin1
 
-#if defined(HAVE_SOLARIS_LARGE_PAGES) && defined(__GNUC__)
-extern "C" int getpagesizes(size_t *, int);
-extern "C" int memcntl(caddr_t, size_t, int, caddr_t, int, int);
-#endif
-
 #ifdef HAVE_FPU_CONTROL_H
 # include <fpu_control.h>
 #elif defined(__i386__)
